Marlin/Marlin_main.cpp

```diff
@@ -4225,6 +4279,128 @@ inline void gcode_M226() {
+          levelProbe_2 = bed_level_probe_pt(ABL_PROBE_PT_2_X - X_PROBE_OFFSET_FROM_EXTRUDER, ABL_PROBE_PT_2_Y - Y_PROBE_OFFSET_FROM_EXTRUDER, current_position[Z_AXIS], verbose_level),
+          levelProbe_3 = bed_level_probe_pt(ABL_PROBE_PT_3_X - X_PROBE_OFFSET_FROM_EXTRUDER, ABL_PROBE_PT_3_Y - Y_PROBE_OFFSET_FROM_EXTRUDER, current_position[Z_AXIS], verbose_level);
+    
+    levelProbe_1 = abs((levelProbe_1 - 5000)/1000);
+    levelProbe_2 = abs((levelProbe_2 - 5000)/1000);
+    levelProbe_3 = abs((levelProbe_3 - 5000)/1000);
+    if (verbose_level > 2) {
+      SERIAL_PROTOCOLPGM("ok ");
+      SERIAL_PROTOCOL_F(levelProbe_1, 10);
+      SERIAL_EOL;
+      SERIAL_PROTOCOLPGM("ok ");
+      SERIAL_PROTOCOL_F(levelProbe_2, 10);
+      SERIAL_EOL;
+      SERIAL_PROTOCOLPGM("ok ");
+      SERIAL_PROTOCOL_F(levelProbe_3, 10);
+      SERIAL_EOL;
+    }
+    clean_up_after_endstop_move();
+    if (!dryrun) {
+      set_bed_level_equation_3pts(levelProbe_1, levelProbe_2, levelProbe_3);
+    }
+
+    #ifndef DELTA
+      if (verbose_level > 0) {
+        plan_bed_level_matrix.debug(" \nBed Level Correction Matrix:");
+      }
+      float x_tmp = current_position[X_AXIS] + X_PROBE_OFFSET_FROM_EXTRUDER,
+            y_tmp = current_position[Y_AXIS] + Y_PROBE_OFFSET_FROM_EXTRUDER,
+            z_tmp = current_position[Z_AXIS],
+            real_z = (float)st_get_position(Z_AXIS) / axis_steps_per_unit[Z_AXIS];
+      apply_rotation_xyz(plan_bed_level_matrix, x_tmp, y_tmp, z_tmp);
+      current_position[Z_AXIS] = z_tmp - real_z + current_position[Z_AXIS];
+      sync_plan_position();
+    #endif // !DELTA
+  }
+#endif
+
+/*
+* M238 - Return ADC value from laser sensor (get distance)
+*/
+uint16_t gcode_M238(uint8_t power) {
+  uint16_t num_samples = 0x0001 << power;
+  uint16_t i = 0;
+  uint32_t sample_sum = 0; // must be 32 bit unsigned int!
+  uint16_t sample_avg = 0;
+
+  // Value must be less than max sample power
+  // (This value was just taken from M234 for simplicity)
+  if(power > ADC_SAMPLE_POWER) {
+    power = ADC_SAMPLE_POWER;
+  }
+  // Take specified amount of readings
+  for(i = 0; i < num_samples; i++) {
+    sample_sum += EXT_ADC_READ_1;
+  }
+  // Take average of sample readings
+  sample_avg = sample_sum >> power;
+
+  return sample_avg;
+}
+
+#ifdef ENABLE_AUTO_BED_LEVELING
+  /*
+  * M239 - Homing and bed leveling combination
+  */
+  inline void gcode_M239() {
+    int verbose_level = code_seen('V') || code_seen('v') ? code_value_short() : 0;
+    if (verbose_level < 0 || verbose_level > 4) {
+      SERIAL_ECHOLNPGM("?(V)erbose Level is implausible (0-4).");
+      return;
+    }
+
+    if (verbose_level > 0) {
+      SERIAL_PROTOCOLPGM("homing device");
+      SERIAL_EOL;
+    }
+    enqueuecommands_P(PSTR("G28"));
+    char cmd[30] = { 0 };
+
+    sprintf_P(cmd, PSTR("M237 V%d"), verbose_level);
+    enqueuecommand(cmd);
+  }
+#endif
+
```

Comment:
I'm not sure we need a function that combines G28 and M237 when we can just issue both of those commands.